src/src/buffer_pool_manager/page_table.rs

```diff
@@ -0,0 +1,48 @@
+use crate::shared::definitions::*;
+use std::collections::HashMap;
+
+#[derive(Debug)]
+pub enum PageState {
+    Clean,
+    Dirty,
+}
+
+#[derive(Debug)]
+pub struct PageTableItem {
+    pub pinned: bool,
+    pub ref_count: usize,
```

Comment:
Why do we need both pinned and ref_count? Doesn't ref_count > 0 imply pinned?

Reply:
`ref_count` is automatically tracked in order to pick a `Page` that will be evicted later. `pinned` is a manual setting that prevents eviction for a `Page` entirely.